modules/distribution-zones/src/integrationTest/java/org/apache/ignite/internal/rebalance/ItRebalanceTest.java

```diff
@@ -146,6 +149,34 @@ void assignmentsChangingOnNodeLeaveNodeJoin() throws Exception {
         );
     }
 
+    @Test
+    @WithSystemProperty(key = THREAD_ASSERTIONS_ENABLED, value = "false")
+    public void compactionAfterAbortedRebalance() {
+        createZone("TEST_ZONE", 1, 1);
+        createTestTable("TEST_TABLE", "TEST_ZONE");
+
+        IgniteImpl igniteImpl = igniteImpl(0);
+
+        TableViewInternal table = unwrapTableViewInternal(igniteImpl.tables().table("TEST_TABLE"));
+
+        MvTableStorage storage = table.internalTable().storage();
+
+        table.keyValueView().put(
+                null,
+                Tuple.create().set("id", 1),
+                Tuple.create().set("val", "value1")
+        );
+
+        CompletableFuture<CompletableFuture<Void>> abortRebalance = storage.startRebalancePartition(0)
+                .thenApply(v -> storage.abortRebalancePartition(0));
+
+        assertThat(abortRebalance, willCompleteSuccessfully());
+
+        CompletableFuture<Void> flush = storage.getMvPartition(0).flush(true);
+
+        assertThat(flush, willCompleteSuccessfully());
```

Comment:
I think that inline of future are enough here.
According to the name of the test, you want to check compaction, but you only check flash, I think something in the test is not finished yet.